Swift/ReplicatorConfiguration.swift

```diff
@@ -124,69 +124,89 @@ public struct ReplicatorConfiguration {
                             userInfo: nil).raise()
                 return
             }
+            
+            _heartbeat = newValue
+        }
+        
+        get {
+            return _heartbeat > 0 ? _heartbeat : ReplicatorConfiguration.defaultHeartbeat
         }
     }
     
-    private var _maxRetries: Int = -1
+    private var _maxAttempts: Int = 0
     /**
      The maximum attempts to perform retry. The retry attempt will be reset when the replicator is
      able to connect and replicate with the remote server again.
      
-     Without setting the maxRetries value, the default maxRetries of 9 times for single shot
+     When setting the  _maxAttempts to zero(default), the default maxAttempts of 10 times for single shot
      replicators and infinite times for continuous replicators will be applied and present to users.
-     Settings the value to 0 will result in no retry attempt.
+     Settings the value to 1 will result in, will perform an initial request and if there is a transient error
+     occurs, will stop will retrying
      
      Setting a negative number will result in InvalidArgumentException being thrown.
      */
-    public var maxRetries: Int {
+    public var maxAttempts: Int {
         set(newValue) {
             guard newValue >= 0 else {
                 NSException(name: .invalidArgumentException,
-                            reason: "Attempt to store negative value in maxRetries",
+                            reason: "Attempt to store negative value in maxAttempts",
                             userInfo: nil).raise()
                 return
             }
             
-            _maxRetries = newValue
+            _maxAttempts = newValue
         }
         
         get {
-            return _maxRetries >= 0
-                ? _maxRetries : self.continuous
-                ? ReplicatorConfiguration.defaultContinousMaxRetries : ReplicatorConfiguration.defaultSingleShotMaxRetries
+            return _maxAttempts > 0
+                ? _maxAttempts : self.continuous
+                ? ReplicatorConfiguration.defaultContinousMaxAttempts : ReplicatorConfiguration.defaultSingleShotMaxAttempts
         }
     }
     
-    /*
-     Max wait time for the next retry.
+    private var _maxAttemptWaitTime: TimeInterval = 0
+    /**
+     Max wait time for the next attempt(retry).
      
      The exponential backoff for calculating the wait time will be used by default and cannot be
-     customized. Set the maxRetryWaitTime to zero or negative value will result in
-     InvalidArgumentException being thrown.
+     customized. Set the value to zero(by default) means using the default max attempts of
+     300 seconds.
+     
+     Set the maxAttemptWaitTime to negative value will result in InvalidArgumentException
+     being thrown.
      */
-    public var maxRetryWaitTime: TimeInterval = 300 {
-        willSet(newValue) {
+    public var maxAttemptWaitTime: TimeInterval {
```

Comment:
I feel like this should be an optional type (for any optional configs). But this can be explored and done as a separate ticket.